filament/include/filament/FilamentAPI.h

```diff
@@ -54,6 +55,22 @@ class UTILS_PUBLIC FilamentAPI {
 template<typename T>
 using BuilderBase = utils::PrivateImplementation<T>;
 
+void builderMakeName(utils::CString& outName, const char* name, size_t len) noexcept;
+
+template <typename Builder>
+class UTILS_PUBLIC BuilderNameMixin {
+public:
+    Builder& name(const char* name, size_t len) noexcept {
+        builderMakeName(mName, name, len);
+        return static_cast<Builder&>(*this);
+    }
+
+    utils::CString getName() const noexcept { return mName; }
```

Comment:
`CString const&` ?

Reply:
if you wanted to be fancy you could even have a version like this:

`utils::CString&& getName() const noexcept & { return mName; }`

that makes the function returns a movable reference if `this` is a temporary :-)